### PR TITLE
Set explicit ExitTimeOut for MacOS agent launchd plist

### PR DESCRIPTION
**Pull request by aleksmaus**
## What does this PR do?

Set explicit ExitTimeOut for MacOS agent launchd plist to 60 secs.
Addresses the issue https://github.com/elastic/elastic-agent/issues/593 where the agent is killed prematurely by launchd without having a chance to shut down gracefully.
See the issue https://github.com/elastic/elastic-agent/issues/593 for more details.

## Why is it important?

Addresses the issue https://github.com/elastic/elastic-agent/issues/593

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas

## How to test this PR locally

The agent should be able to stop without leaving any child processes running. The agent should not receive SIGKILL from launchd. Refer to the issue for details https://github.com/elastic/elastic-agent/issues/593

## Related issues

- Closes https://github.com/elastic/elastic-agent/issues/593

## Screenshots
The updated agent ```/Library/LaunchDaemons/co.elastic.elastic-agent.plist``` installed with ```ExitTimeOut```
Screen Shot 2022-06-22 at 12 03 11 PM

## Logs
Launchd log after the change:
Screen Shot 2022-06-22 at 12 06 08 PM

### CODE REVIEW

**Review by ph — APPROVED** [review]
This LGTM, thanks for the details screenshot and all. @michel-laterman Can you do another review on this?

File: internal/pkg/agent/install/svc.go [hunk 1]
```diff
@@ -18,6 +19,12 @@ const (
 
 	// ServiceDescription is the description for the service.
 	ServiceDescription = "Elastic Agent is a unified agent to observe, monitor and protect your system."
+
+	// Set the launch daemon ExitTimeOut to 60 seconds in order to allow the agent to shutdown gracefully
+	// At the moment the version 8.3 & 8.4 of the agent are taking about 11 secs to shutdown
+	// and the launchd sends SIGKILL after 5 secs which causes the beats processes to be left running orphant
```
Comment (michel-laterman) [inline internal/pkg/agent/install/svc.go:25]:
```suggestion
	// and the launchd sends SIGKILL after 5 secs which causes the beats processes to be left running orphaned
```

File: internal/pkg/agent/install/svc.go [hunk 2]
```diff
@@ -45,5 +52,57 @@ func newService() (service.Service, error) {
...
+		cfg.Option["LaunchdConfig"] = darwinLaunchdConfig
+		cfg.Option["ExitTimeOut"] = darwinServiceExitTimeout
+	}
+
+	return service.New(nil, cfg)
 }
+
+// A copy of the launchd plist template from github.com/kardianos/service
+// with added .Config.Option.ExitTimeOut option
+var darwinLaunchdConfig = `<?xml version='1.0' encoding='UTF-8'?>
```
Comment (michel-laterman) [inline internal/pkg/agent/install/svc.go:70]: Should this be a `const`?

**Review by michel-laterman — APPROVED** [review]
lgtm, just have some nitpicks

**Review by james-elastic — APPROVED** [review]

File: internal/pkg/agent/install/svc.go (hunk 2 above)
Comment (aleksmaus) [reply internal/pkg/agent/install/svc.go:70]: yep. the original lib had it as var heh

**Pull request merged**